crawler/crawler.go

```diff
@@ -109,6 +104,9 @@ func checkAvailability(repository Repository, processedCounter prometheus.Counte
 			log.Warn("Validator errors:" + err.Error())
 		}
 	}
+
+	// Defer waiting group close.
+	wg.Done()
```

Comment:
maybe a `defer wg.Done()` at the beginning of the function is more safe

Reply:
defer is called after the surrounding function returns (https://blog.golang.org/defer-panic-and-recover). As goroutine there is no return in function (only channel communication). The defer will never be called.

Reply:
yes, correct